pkg/services/ngalert/api/tooling/definitions/alertmanager_validation.go

```diff
@@ -87,6 +81,21 @@ func (t *NotificationTemplate) Validate() error {
 	}
 	t.Template = content
 
+	// Validate template contents. We try to stick as close to what will actually happen when the templates are parsed
+	// by the alertmanager as possible. That means parsing with both the text and html parsers and making sure we set
+	// the template name and options.
+	ttext := tmpltext.New(t.Name).Option("missingkey=zero")
+	ttext.Funcs(tmpltext.FuncMap(template.DefaultFuncs))
+	if _, err := ttext.Parse(t.Template); err != nil {
+		return fmt.Errorf("invalid template: %w", err)
+	}
+
+	thtml := tmplhtml.New(t.Name).Option("missingkey=zero")
```

Comment:
This surprises me, I thought the Alertmanager didn't support HTML templates?

Reply:
Surprised me as well, maybe I'm misunderstanding somehow but I'm mostly just going off of this code in the alertmanager which is ran when parsing templates: https://github.com/grafana/prometheus-alertmanager/blob/main/template/template.go#L93-L98

Reply:
Alertmanager does - Grafana doesn't. Grafana's notifiers are where the HTML is disallowed.

Reply:
If it fails here, it will also fail when the templates are parsed by our alertmanager since we do the same TmplHtml parsing then: https://github.com/grafana/prometheus-alertmanager/blob/main/template/template.go#L93-L98

Reply:
Also, the current code **only** uses `TmplHtml`